cpp/src/link_analysis/pagerank.cu

```diff
@@ -114,7 +115,12 @@ int pagerank(IndexType n, IndexType e, IndexType *cscPtr, IndexType *cscInd, Val
     copy(n, pagerank_vector, tmp);
   }
 
-  fill(n, b, randomProbability);
+  if (has_personalization) {
+    fill(n, b, static_cast<ValueType>(0));
+    scatter(prsLen, prsVal, b, prsVtx);

```

Comment:
Should `b` sum to one from the mathematical perspective of the problem? 
If so, what if the user's values don't? Should we normalize by nrm1 when it is larger or fill the rest of the vector with the correct uniform value when it is smaller? We could also return an error.

Reply:
Discussed on slack with Aatish and Haekyu. We may normalize the input.

```
if |Q| == 1.0: continue
else if Q == 0.0 <- uniform dist for all nodes
else Q <- Q / |Q| 
```

Reply:
Fixed [here](https://github.com/rapidsai/cugraph/pull/300/files?file-filters%5B%5D=.cu&file-filters%5B%5D=.py&owned-by%5B%5D=#diff-c774f1d383f54e52fdc1f235c24c0251R123)